### PR TITLE
Allow Bundle IDs that have a valid prefix - strict validation

### PR DESCRIPTION
**Pull request by maksymmalyhin**
This is an updated version of #2430 to address comments:
- strict validation for extension bundle IDs

### CODE REVIEW

File: Firebase/Core/FIRBundleUtil.m [hunk 1]
```diff
@@ -45,13 +47,33 @@ + (NSArray *)relevantURLSchemes {
...
+
+    if ([applicationBundleIdentifier isEqualToString:bundle.bundleIdentifier]) {
       return YES;
     }
   }
   return NO;
 }
 
++ (NSString *)bundleIdentifierByRemovingLastPartFrom:(NSString *)bundleIdentifier {
+  NSString *bundleIdComponentsSeparator = @".";
```
Comment (ryanwilson) [inline Firebase/Core/FIRBundleUtil.m]: Nit: `bundleID` instead of `Id` here and below.

File: Firebase/Core/FIRBundleUtil.m [hunk 2]
```diff
@@ -45,13 +47,33 @@ + (NSArray *)relevantURLSchemes {
...
+  NSArray<NSString *> *bundleIdComponents =
+      [bundleIdentifier componentsSeparatedByString:bundleIdComponentsSeparator];
+  if (bundleIdComponents.count < 2) {
+    return @"";
+  }
+
+  NSMutableArray<NSString *> *mutableNundleIdComponents = [bundleIdComponents mutableCopy];
```
Comment (ryanwilson) [inline Firebase/Core/FIRBundleUtil.m]: Nit: `s/Nundle/Bundle/` here and below.

File: Firebase/Core/Private/FIRBundleUtil.h [hunk 3]
```diff
@@ -45,8 +45,8 @@
 + (NSArray *)relevantURLSchemes;
 
 /**
- * Checks if the bundle identifier exists in the given bundles.
+ * Checks the given bundles to see of them is a prefix of the given identifier.
```
Comment (ryanwilson) [inline Firebase/Core/Private/FIRBundleUtil.h]: Maybe "Checks if any of the given bundles have a matching bundle identifier prefix (removing extension suffixes)."?

File: FirebaseCore.podspec [hunk 4]
```diff
@@ -29,6 +29,7 @@ Firebase Core includes FIRApp and FIROptions which provide central configuration
   s.private_header_files = 'Firebase/Core/Private/*.h'
   s.framework = 'Foundation'
   s.dependency 'GoogleUtilities/Logger', '~> 5.2'
+  s.dependency 'GoogleUtilities/Environment', '~> 5.2'
```
Comment (maksymmalyhin) [inline FirebaseCore.podspec]: @ryanwilson Not sure we catch missing dependencies in our tests...

Comment (ryanwilson) [reply FirebaseCore.podspec]: Nice catch - I'm not sure why this wouldn't be caught by `pod lib lint` - @paulb777  any ideas?

Comment (paulb777) [reply FirebaseCore.podspec]: Because it's always a transitive dep - https://github.com/firebase/firebase-ios-sdk/blob/master/GoogleUtilities.podspec#L38

Comment (ryanwilson) [reply FirebaseCore.podspec]: Makes sense! Can you please move it up one to be above `Logger` (alphabetical order)?

**Pull request merged**